corehq/util/metrics/views.py

```diff
@@ -0,0 +1,19 @@
+import os
+
+import prometheus_client
+from django.http import HttpResponse
+from prometheus_client import multiprocess
+
+
+def prometheus_metrics(request):
+    """Exports /metrics as a Django view.
```

Comment:
What is this view used for? Should it be protected by an admin-only decorator? I'm assuming this is not the only interface for viewing prometheus metrics... seems like it would be bad if metrics became unavailable if this django view became inaccessible for whatever reason.

Reply:
Because Prometheus is primarily a pull based system the metrics need to be exposed as a view. There are a few options for this:
1. As a Django view
    * this can either be protected by a shared secret / basic auth or, since it will be behind nginx in production we can deny access to it via the proxy.
2. A separate HTTP server run on Django startup
    * `prometheus_client.start_http_server(port, addr=addr)`
3. A completely separate HTTP server

I'm leaning towards 3: f1874e43522

Reply:
Ah, that makes sense.

> 1. As a Django view  ... since it will be behind nginx in production we can deny access to it via the proxy

I'm -1 on this option if it's not authenticated in some way. It sounds like it may increase the complexity of proxy configuration, and someone who is setting up their own instance of HQ but doesn't know that it should be hidden may inadvertently advertise their metrics to the world.

Reply:
If we went that route I would update commcare-cloud to have the correct configuration by default. But I think for now I'll remove this view or restrict it's access.

Reply:
https://github.com/dimagi/commcare-hq/pull/26903